### PR TITLE
release-1.16 Pick require affinity for typha

### PR DESCRIPTION
**Pull request by freecaykes**
## Description
pick from https://github.com/tigera/operator/pull/1308 for release-v1.16

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [x] Milestone set according to targeted release.
- [x] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/controller/installation/core_controller.go
Comment (tmjd) [inline pkg/controller/installation/core_controller.go:464]: This should not be part of the pick, it wasn't in #1308

**Review by tmjd — CHANGES_REQUESTED** [review]
How did you pick these changes? I'm asking because of the addition in core_controller.go Line 457-464.
If you performed the pick by picking #1308 I think instead you should pick the 3 individual commits that were picked as part of #1308 instead of picking #1308.
I didn't see any other changes that were not changes in the original pick but I'm concerned I might have missed something.

File: pkg/controller/installation/core_controller.go
Comment (freecaykes) [reply pkg/controller/installation/core_controller.go]: I remember that's what I did, cherry picked the individual commits frin #1308 , this was part of the PR  https://github.com/tigera/operator/pull/1286 that was in https://github.com/tigera/operator/pull/1308, maybe a bad merge 

removing it here https://github.com/tigera/operator/pull/1368/commits/3ced8136051e91c7bc6762a64d45b3a567dc5423

**Review by tmjd — APPROVED** [review]
LGTM

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
